### ISSUE TITLE
Valetudo remembers old MQTT identifier

### ISSUE THREAD

**Issue by alve89** [Hypfer/Valetudo#1329]
**First of all: Thank you - you're doing a really great job! I flashed my S6 today with [Dustbuilder](https://dustbuilder.dontvacuum.me) and I'm fascinated how properly the Valetudo software itself is working (including the map, rooms and so on). Both thumbs up!**

But now to my problem.

## Describe the bug
After setting up the MQTT settings the generic identifier is still used.

That's the generic identifier:
![image](https://user-images.githubusercontent.com/11487054/148097225-c7f8aeef-5df0-4926-88a6-08102c02350b.png)

That's the identifier is used:
![image](https://user-images.githubusercontent.com/11487054/148097337-ddd0020a-5bda-4202-9611-4fbd1b10b057.png)

That's what MQTT Explorer recognizes:
![image](https://user-images.githubusercontent.com/11487054/148097443-c176823e-7bee-4f5d-ad93-7733795ee9d7.png)

That's what my MQTT broker (Mosquitto as Home Assistant add-on in HassOS) logs:
![image](https://user-images.githubusercontent.com/11487054/148100105-6f959494-709a-4080-a674-5c2454d4732a.png)

In my opinion `valetudo_ThatLightheartedCurlew` shouldn't be used anymore because I replaced it by `myVeryPrivateRoborock`...?

These are the steps I tried to solve the problem (multiple times, in different order):
- Restart Mosquitto
- Restart Home Assistant
- Restart Roborock (via SSH)
- Changing MQTT login credentials
- Changing MQTT identifier

Valetudo or Mosquitto still use the identifier `valetudo_ThatLightheartedCurlew` although it was changed to `myVeryPrivateRoborock`.

### Screenshots

This is my Valetudo MQTT configuration:
![image](https://user-images.githubusercontent.com/11487054/148099387-d55c3b99-c499-43a6-a2f7-4be54c6e96cf.png)

### Vacuum Model

Roborock S6

### Valetudo Version

![image](https://user-images.githubusercontent.com/11487054/148099471-6bc453b9-2604-4bdf-8e34-58f7e6261603.png)

## Expected behavior

I expect to see the same identifier (`myVeryPrivateRoborock`) everywhere and especially I don't see `valetudo_ThatLightheartedCurlew` somewhere after changing it.

## Additional context

I just rebooted the Roborock once more (after completely rebooting the whole RPi):
![image](https://user-images.githubusercontent.com/11487054/148099879-9b0a3372-c277-4f6c-b204-a009aaad2fa3.png)

This shows that Valetudo is "logging in" to MQTT as `valetudo_ThatLightheartedCurlew`.

Could this also be the reason why the [Valetudo Map Card](https://github.com/TheLastProject/lovelace-valetudo-map-card) doesn't work?
![image](https://user-images.githubusercontent.com/11487054/148100379-14a0333f-ef9e-4bab-b8c2-c75cf64f4ebe.png)
![image](https://user-images.githubusercontent.com/11487054/148100394-82fbe9f6-ef68-465a-bf42-f500486865c3.png)

**Comment by Hypfer** [comment]
Are you sure that you're not just seeing retained messages?

**Comment by Hypfer** [comment]
TL;DR:

The `MQTT ClientId` is unaffected by the identifier setting.

https://github.com/Hypfer/Valetudo/blob/5bab773230a4d51761b5ea3a9d1c274e4a94bbab/backend/lib/mqtt/MqttController.js#L122-L123

This is indeed true. I don't think it should be though, given that it only appears in the mqtt broker logs.
It might be a misunderstanding of the meaning of identifier.  The machine readable identifier is currently used to e.g. name the topics, name the home assistant device etc.

Problem is that the `MQTT ClientId` has to be unique as the mqtt broker will terminate the older connection if another device with the same client id appears because it assumes that it's the same device and the connection just broke

There will be at least three people misconfiguring this so that multiple robots fight over the same identifier. Then, there will be at least five issues complaining about Valetudos MQTT client being broken.



> Could this also be the reason why the Valetudo Map Card doesn't work?

No

**Comment by alve89** [comment]
Alright, thank you for this detailed explanation. 
So this is an expected behavior as

- `valetudo_ThatLightheartedCurlew` is the `clientId` (which is auto-generated) and
- `myVeryPrivateRoborock` is only used for topics and so on.

Do you also know which of both have to be used for the Lovelace Map Card?
If yes, your answer will be very appreciated, if no, please just close this issue. 

Thanks again for your effort and help!

**Comment by alve89** [comment]
I found the problem with the Lovelace Map Card. 
The configuration is
```
type: custom:valetudo-map-card
vacuum: valetudo_myveryprivateroborock
```

So it's the user-given robot name in **lowercase**.
